### PR TITLE
Fix win dev server and improve webpack configuration

### PR DESCRIPTION
**Pull request by BeatPlus**
This PR aims to fix the dev process that sometimes used to fail. Now executes the electron async.

It also improves the webpack commands by splitting into two (main + renderer)

### CODE REVIEW

File: src/main/index.js [hunk 1]
```diff
@@ -95,15 +103,10 @@ function createMainWindow() {
...
+app.on('ready', async () => {
   if (process.env.NAME === 'development') {
     /* eslint-disable no-console */
-    [REDUX_DEVTOOLS, REACT_DEVELOPER_TOOLS]
-      .forEach(extension =>
-        installExtension(extension)
-          .then((name) => console.log(`Added Extension:  ${name}`))
-          .catch((err) => console.log('An error occurred: ', err))
-      );
+    await installExtensions();
```
Comment (kontrollanten) [inline src/main/index.js]: Nice!

File: src/main/index.js [hunk 2]
```diff
@@ -95,15 +103,10 @@ function createMainWindow() {
   return win;
 }
 
-app.on('ready', () => {
+app.on('ready', async () => {
   if (process.env.NAME === 'development') {
     /* eslint-disable no-console */
```
Comment (kontrollanten) [inline src/main/index.js]:
```suggestion
```

File: src/main/index.js [hunk 3]
```diff
@@ -95,15 +103,10 @@ function createMainWindow() {
...
-    [REDUX_DEVTOOLS, REACT_DEVELOPER_TOOLS]
-      .forEach(extension =>
-        installExtension(extension)
-          .then((name) => console.log(`Added Extension:  ${name}`))
-          .catch((err) => console.log('An error occurred: ', err))
-      );
+    await installExtensions();
     /* eslint-enable no-console */
```
Comment (kontrollanten) [inline src/main/index.js]:
```suggestion
```

File: src/main/index.js [hunk 4]
```diff
@@ -6,12 +6,20 @@ import { migrateSettings } from './migrate-settings';
...
 import getMenu from './get-menu';
-import installExtension, { REACT_DEVELOPER_TOOLS, REDUX_DEVTOOLS } from 'electron-devtools-installer';
 
 const settings = require('electron-settings');
-
 require('electron-dl')({saveAs: true});
 
+const installExtensions = async () => {
+  const installer = require('electron-devtools-installer');
+  const forceDownload = !!process.env.UPGRADE_EXTENSIONS;
```
Comment (kontrollanten) [inline src/main/index.js]: Do the developer needs to set the env variable by himself? Or is it some automagic that will set this?

**Comment by kontrollanten** [comment]
Great work!

File: src/main/index.js (hunk 4 above)
Comment (BeatPlus) [reply src/main/index.js]: The developer needs to set the env only if he wants to force updates [see DevTools](https://github.com/electron-react-boilerplate/electron-react-boilerplate/wiki/DevTools).

```
$ UPGRADE_EXTENSIONS=1 yarn run dev
# For Windows
$ set UPGRADE_EXTENSIONS=1 && yarn run dev```

**Pull request merged**